### ISSUE TITLE
Can't get it compiled.

### ISSUE THREAD

**Issue by guijun** [Loki-Astari/ThorsSerializer#10]
I cloned this repo and tried : make -f MakeSample
it's strange that I many compile error appeared and make failed.
I'm use xubuntu 14.04

**Comment by Loki-Astari** [comment]
Just pushed to master a new major update.
The build is now done using the standard autotools model.

./configure
make
sudo make install

The configuration script does a lot more validation of the environment than the previous system. Please let me know if you have any questions.

**Comment by lordnynex** [comment]
Hello!

Sorry to dig up an old bug. I've been using your JsonSerializer (Parser) as a learning experience for flex/yacc. I did so based on a codereview post you did several years ago. If you don't mind, what was the original issue here? It's hard to tell from the addition of the build submodule how a vanilla Makefile would be changed to fix this. 

This is a very useful project. Thanks for writing it.

**Comment by Loki-Astari** [comment]
The problem was caused because people were using an older version of `bison` on their systems than I was using in mine.

The solution on my end is to explicitly check the version of bison and require a minimum version. This is one of the reasons I moved the build system to use autoconfig system rather than the simple makefile.


TO build now:

    ./configure                             # Checks that all the tools are at required version
    make
    sudo make install

**Comment by lordnynex** [comment]
Not at all what I was asking, but thanks for the reply anyway.

**Comment by Loki-Astari** [comment]
Not sure how to answer then.

But if you are asking how to make the Simple Makefile work. I decided that was a problem I did not want to solve (because there are tools that already do that (automake).

I am sure it can be done but basically you will have to modify your make file to get the version of bison currently installed and generate an error if if is blow the minimum requirements.

**Comment by lordnynex** [comment]
Well after some digging, I found that a lot has changed with Bison since your original posting in 2012. I think the original post was either incomplete or missing some interface implementations that where not there in in 2012. I thought maybe I was missing a compiler flag. 

After reading a bit more yesterday I realized there are a few different ways used now to generate c++ lexers. Comparing to ThorSerializer the project now, theres quite a few differences so I decided to stop using the original as an example.
